app/mirage/config.js

```diff
@@ -207,6 +207,31 @@ export function testConfig() {
         return response;
     });
 
+    this.get('/posts/:id', function (db, request) {
+        let {id} = request.params;
+        let post = db.posts.find(id);
+
+        return {
+            posts: [post]
+        };
+    });
+
+    this.put('/posts/:id', function (db, request) {
+        let [attrs] = JSON.parse(request.requestBody).posts;
+        let post;
+
+        if (isBlank(attrs.slug) && !isBlank(attrs.title)) {
+            attrs.slug = attrs.title.dasherize();
+        }
+
+        // NOTE: this does not use the post factory to fill in blank fields
+        post = db.posts.insert(attrs);
```

Comment:
As this is a `put` request we want to be updating the post rather than inserting a new one. Example following the `this.put('/tags/:id', ...` route further down:

```
        let {id} = request.params;
        let [attrs] = JSON.parse(request.requestBody).tags;
        let post = db.posts.update(id, attrs);

        return {
            posts: [post]
        };
```